pkg/ansible/watches/watches.go

```diff
@@ -448,3 +455,32 @@ func getIntegerEnvWithDefault(envVar string, defValue int) int {
 	}
 	return val
 }
+
+// getIntegerEnvMaxReconcile looks for gloabal variable "MAX_CONCURRENT_RECONCILES_<group>_<kind>",
+// if not present it checks for "WORKER_<group>_<kind>" and logs deprecation message
+// if required. If both of them are not set, we use the default value passed on by command line
+// flags.
+func getIntegerEnvMaxReconcile(envVarMaxWorker, envVarMaxReconciler string, defValue int) int {
+	val := defValue
+	if envValRecon, ok := os.LookupEnv(envVarMaxReconciler); ok {
+		if i, err := strconv.Atoi(envValRecon); err != nil {
+			log.Info("Could not parse environment variable as an integer; using default value",
+				"envVar", envVarMaxReconciler, "default", defValue)
+		} else {
+			val = i
+		}
+	} else if !ok {
+		if envValWorker, ok := os.LookupEnv(envVarMaxWorker); ok {
+			log.Info("Deprecating gloabal variable WORKER_<group>_<kind>, use " +
+				"MAX_CONCURRENT_RECONCILES_<group>_<kind> instead")
```

Comment:
We should use the actual environment variable names. Something like this:
```suggestion
			deprecationMsg := fmt.Sprintf("Environment variable %s is deprecated, use %s instead", envVarMaxWorker, envVarMaxReconciler)
			log.Info(deprecationMsg)
```